### PR TITLE
Attempted bugfix

### PR DESCRIPTION
**Pull request by Sotskin**
Do not merge.
Attempted bugfix for illegal memory access problem that happens when running with ThreadedEngine. Not effective.

### CODE REVIEW

File: src/executor/graph_executor.cc [hunk 1]
```diff
@@ -1428,18 +1429,28 @@ void GraphExecutor::InitCachedOps() {
...
       exec->Run(ctx, is_gpu);
+      std::cout<<"RunF 2"<<std::endl;
       // call on complete only if it is async op
       if (!is_async) {
         if (is_gpu) {
         #if MXNET_USE_CUDA
           // Wait GPU kernel to finish.
+          std::cout<<"RunF 3"<<std::endl;
           ctx.get_stream<gpu>()->Wait();
+          Swap::Get()->UnlockSwap();
```
Comment (fegin) [inline src/executor/graph_executor.cc]: Move the UnlockSwap() after on_complete() (Line 1450).

File: src/executor/graph_executor.cc [hunk 2]
```diff
@@ -1672,18 +1683,29 @@ GraphExecutor::CachedSegOpr GraphExecutor::CreateCachedSegOpr(size_t topo_start,
...
       exec->Run(ctx, is_gpu);
+      std::cout<<"RunF2 1.8"<<std::endl;
     }
+    std::cout<<"RunF2 2"<<std::endl;
     if (is_gpu) {
 #if MXNET_USE_CUDA
       // Wait GPU kernel to finish.
+      std::cout<<"RunF2 3"<<std::endl;
       ctx.get_stream<gpu>()->Wait();
+      Swap::Get()->UnlockSwap();
```
Comment (fegin) [inline src/executor/graph_executor.cc]: Move this UnlockSwap() after on_complete() (Line 1707).

**Comment by Sotskin** [comment]
As for the new commit LRU can pass 10/10 times. This is achieved by preventing any handle smaller than 20Mb to from being swapped out. So it is not a true fix.
NaiveHistory algoirthm also appears to be running just fine, (passed 5/5 times), but unlock need to be moved immediately after Wait()

**Comment by fegin** [comment]
I'll take a look at the issue.

**Pull request merged**